lib/mongo-object.test.js

```diff
@@ -201,8 +201,21 @@ describe('MongoObject', () => {
   });
 
   it('makeKeyGeneric', () => {
-    const generic = MongoObject.makeKeyGeneric('foo.0.0.ab.c.123.4square.d.67e.f.g.1');
-    expect(generic).toEqual('foo.$.$.ab.c.$.4square.d.67e.f.g.$');
+    function testMakeKeyGeneric(input, expectedOutput) {
+      const generic = MongoObject.makeKeyGeneric(input);
+      expect(generic).toEqual(expectedOutput);
+    }
+    testMakeKeyGeneric(0, null);
+    testMakeKeyGeneric({}, null);
+    testMakeKeyGeneric('foo', 'foo');
+    testMakeKeyGeneric('foo.bar', 'foo.bar');
+    testMakeKeyGeneric('foo.$', 'foo.$');
+    testMakeKeyGeneric(
+      'foo.0.0.ab.c.123.4square.d.67e.f.g.1',
+      'foo.$.$.ab.c.$.4square.d.67e.f.g.$'
+    );
+    testMakeKeyGeneric('foo.$[].foo.$[bar].$.$[]', 'foo.$.foo.$.$.$');
+    testMakeKeyGeneric('foo.$foo.$foo[bar]foo', 'foo.$foo.$foo[bar]foo');
```

Comment:
Maybe also test with `foo$[]` and `foo$[bar]` keys (where dollar sign isn't first character). Those should be kept as is, too.